recaf-launcher/src/main/java/dev/xdark/recaf/Launcher.java

```diff
@@ -0,0 +1,251 @@
+package dev.xdark.recaf;
+
+import com.google.gson.GsonBuilder;
+import dev.dirs.BaseDirectories;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.PrintStream;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.jar.JarFile;
+import java.util.jar.Manifest;
+import joptsimple.OptionParser;
+import joptsimple.OptionSet;
+import joptsimple.OptionSpec;
+import org.apache.commons.io.IOUtils;
+
+/**
+ * Recaf's launcher entrypoint.
+ *
+ * @author xDark
+ */
+public final class Launcher {
+
+  private static final String API_URL = "https://api.github.com/repos/Col-E/Recaf/releases/latest";
+  private static final String ISSUES_URL = "https://github.com/Col-E/Recaf/issues";
+
+  private static final List<String> UPDATE_FAIL = Collections.singletonList("--updateFailed");
+  private static final PrintStream PS = System.err;
```

Comment:
I'll be using [logback](https://mvnrepository.com/artifact/ch.qos.logback/logback-classic/1.2.3) in the core for configurable logging, can use that here as well. It can replace the logging wrapper code below.

The default format of messages can be changed in a config file if you want to retain the same visuals of using this.